### ISSUE TITLE
Update to newer kernel

### ISSUE THREAD

**Issue by cmur2** [raspberrypi/gpioexpander#1]
Is it possible to update to a more recent kernel like the Linux 4.14 kernel available at https://github.com/raspberrypi/linux/tree/rpi-4.14.y (via https://www.raspberrypi.org/forums/viewtopic.php?f=29&t=197689)? 

Would it be enough to change this line https://github.com/raspberrypi/gpioexpander/blob/master/gpioexpand/configs/gpioexpand_defconfig#L13 or are there more changes needed?

**Comment by maxnet** [comment]
>Is it possible to update to a more recent kernel like the Linux 4.14 kernel

Technically, yes.
But newer isn't always better, though.

>Would it be enough to change this line >https://github.com/raspberrypi/gpioexpander/blob/master/gpioexpand/configs/gpioexpand_defconfig#L13

You will need to update the kernel configuration as well: https://github.com/raspberrypi/gpioexpander/blob/master/gpioexpand/board/kernelconfig-recovery.armv6
Can probably borrow the file from the noobs repo.

And may need to update the rpi-firmware version: https://github.com/raspberrypi/gpioexpander/blob/master/gpioexpand/package/rpi-firmware-custom/rpi-firmware-custom.mk#L7

**Comment by cmur2** [comment]
>  But newer isn't always better, though.

Yes I agree to that and since I do not track kernel regressions in context of Raspberry Pi it could very well be the case. Do you know of any specific drawbacks affecting this project? (btw thanks for this really amazing project it really offered me a new perspective and use case!)

I will probably try updating following your pointers. I [read](https://pinout.xyz/pinout/1_wire) that starting with Linux 4.9.28 there is the ability to dynamically load device tree overlays allowing e.g. easy setup of multiple 1-wire busses which I want to try with my Pi Zero (and a couple of other stuff).

**Comment by maxnet** [comment]
Do note that `pigpiod` that is offered through this image does not have any use for such device tree overlays.
Those are mainly used for tieing pins to kernel modules, and pigpiod does not rely on kernel modules but talks to hardware directly.
